Library/Formula/wellington.rb

```diff
@@ -17,35 +17,58 @@ class Wellington < Formula
 
   depends_on "go" => :build
   depends_on "pkg-config" => :build
-  depends_on "autoconf" => :build
-  depends_on "automake" => :build
-  depends_on "libtool" => :build
 
-  go_resource "github.com/wellington/spritewell" do
-    url "https://github.com/wellington/spritewell.git",
-        :revision => "748bfe956f31c257605c304b41a0525a4487d17d"
+  stable do
+    depends_on "libsass"
   end
 
-  go_resource "github.com/go-fsnotify/fsnotify" do
-    url "https://github.com/go-fsnotify/fsnotify.git",
-        :revision => "f582d920d11386e8ae15227bb5933a8f9b4c3dec"
+  devel do
+    url "https://github.com/wellington/wellington/archive/c574754c39806e8c52682f15f49c8ff819d0f962.tar.gz"
+    sha256 "92c5d4bdd6260f96aed0e20fa06c6200b1c57281dbd32dbd00b1c06636ef6e10"
+
+    depends_on "autoconf" => :build
+    depends_on "automake" => :build
+    depends_on "libtool" => :build
+  end
+
+  go_resource "github.com/tools/godep" do
+    url "https://github.com/tools/godep.git", :revision => "58d90f262c13357d3203e67a33c6f7a9382f9223"
+  end
+
+  go_resource "github.com/kr/fs" do
+    url "https://github.com/kr/fs.git", :revision => "2788f0dbd16903de03cb8186e5c7d97b69ad387b"
+  end
+
+  go_resource "golang.org/x/tools" do
+    url "https://github.com/golang/tools.git", :revision => "473fd854f8276c0b22f17fb458aa8f1a0e2cf5f5"
   end
 
   def install
-    ENV.cxx11
-    # go_resource doesn't support gopkg, do it manually then symlink
-    mkdir_p buildpath/"src/gopkg.in"
-    ln_s buildpath/"src/github.com/go-fsnotify/fsnotify",
-         buildpath/"src/gopkg.in/fsnotify.v1"
-    ENV["PKG_CONFIG_PATH"] = buildpath/"libsass/lib/pkgconfig"
+    ENV["VERSION"] = File.read("version.txt").chomp
+    ENV["GOPATH"] = buildpath
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd "src/github.com/tools/godep" do
+      system "go", "install"
+    end
+    system "bin/godep", "restore"
+
+    # Build libsass from source for devel build
+    if build.devel?
+      ENV.cxx11
+      ENV["PKG_CONFIG_PATH"] = buildpath/"src/github.com/wellington/go-libsass/lib/pkgconfig"
+
+      ENV.append "CGO_LDFLAGS", "-stdlib=libc++" if ENV.compiler == :clang
+      cd "src/github.com/wellington/go-libsass/" do
+        system "make", "deps"
+      end
+    end
+
+    # symlink into $GOPATH so builds work
     mkdir_p buildpath/"src/github.com/wellington"
     ln_s buildpath, buildpath/"src/github.com/wellington/wellington"
-    Language::Go.stage_deps resources, buildpath/"src"
-    ENV["GOPATH"] = buildpath
-    ENV.append "CGO_LDFLAGS", "-stdlib=libc++" if ENV.compiler == :clang
-    system "make", "deps"
-    system "go", "build", "-x", "-v", "-o", "dist/wt", "wt/main.go"
 
+    system "go build -ldflags \"-X main.version #{ENV["VERSION"]}\" -o dist/wt wt/main.go"
```

Comment:
"-X main.Version ..." is one argument, so it sounds like the version.txt
needs to be safely stored in a ruby variable. version.txt is versioned and
so modifying it would be difficult. I'm not a homebrew expert, is there a
preferred way to escape the file contents so no arbitrary code can be
executed from with it. Then this line can be modified to use the escaped
variable reducing the risk noted above.
On Mon, May 18, 2015 at 6:56 AM Mike McQuaid <notifications@github.com>
wrote:

> In Library/Formula/wellington.rb
> <https://github.com/Homebrew/homebrew/pull/39195#discussion_r30500822>:
>
> >
> > +    system "go build -ldflags \"-X main.version #{ENV["VERSION"]}\" -o dist/wt wt/main.go"
>
> They need separated because otherwise you allow arbitrary command
> execution from the version.txt file.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Homebrew/homebrew/pull/39195/files#r30500822>.
>